naas/models/subscriber_email_address.py

```diff
@@ -35,15 +37,21 @@ def confirmation_code(self):
 
     def confirmed_at(self):
         """Returns the confirmed at timestamp value"""
-        return self.attributes.get('confirmed_at')
+        try:
+          return datetime.datetime.strptime(
+            self.attributes.get('confirmed_at'), '%Y-%m-%dT%H:%M:%S%z')
+        except TypeError as e:
+          None
```

Comment:
@arthuroe - This is an example of an _optional_ timestamp. An email address might not yet be _confirmed_, and so this would be `null` in the database. I tested using a record that has a `null` entry in the DB. As expected, it raised the `TypeError`, which we want to suppress here.

The other option is that we have a separate attribute of `confirmed_at_value` - which is the representation that comes from the API. Then we check to see if that is `None`. If it is not, then we coerce to the `datetime`.

Thoughts?

Reply:
@nateklaiber I think the latter could do.
```
def confirmed_at(self):
    if self.attributes.get('confirmed_at'):
        return datetime.datetime.strptime( 
                self.attributes.get('confirmed_at'), '%Y-%m-%dT%H:%M:%S%z')
```
`None` is implicitly returned where the check is not passed.

Reply:
The main reason I use exception handling in the `ruby` gem is to handle the case where an invalid value is passed for one of the dates. I use test cases like passing in `nil`, `random` (as a strong), or `23` (as an Integer). The exception handling captures any parse error and returns `nil` (`None`, here), whereas a condition check could lead to a false positive. `self.attributes.get('confirmed_at')` might have a value that is not parse-able by `strptime`. 

Technically this should never happen, as they will always be built from the API - but when I build unit tests I try and protect from any situation. 

I'm good to follow your lead here, just wanted to provide some more context.